packages/graphback-datasync/src/scheduler.ts

```diff
@@ -0,0 +1,76 @@
+import * as Agenda from "agenda";
+import { Db, MongoClient } from "mongodb";
+
+export interface PruneConfiguration {
+  agendaConfig?: Agenda.AgendaConfiguration
+  pruneInterval?: string | number
+}
+
+export interface TTLMap {
+  [modelName: string]: number
+}
+
+/**
+ * A class for creating prune jobs
+ */
+export class PruneScheduler {
+  public objectDb: Db;
+  public pruneInterval: string | number;
+  public agenda: Agenda;
+
+  public constructor(config: PruneConfiguration, db: Db) {
+    this.agenda = new Agenda(config.agendaConfig);
+    if (config.agendaConfig?.db === undefined && config.agendaConfig?.mongo === undefined) {
+      this.agenda.mongo(db);
+    }
```

Comment:
This will create a collection in the db to track scheduled jobs, which is okay since we want the jobs to be distributed jobs and be run once even if we have multiple instances. Does the lib take care of wiping out old scheduled jobs?

Reply:
Nope, we have to do that ourselves using [purge](https://www.npmjs.com/package/agenda#purge)

Reply:
I haven't added that functionality yet, will try something in next push regarding this